crates/core/src/transport/connection_handler.rs

```diff
@@ -652,13 +653,13 @@ enum ConnectionEvent {
 }
 
 struct InboundRemoteConnection {
-    inbound_packet_sender: mpsc::Sender<PacketData>,
-    inbound_intro_packet: Option<PacketData>,
+    inbound_packet_sender: mpsc::Sender<PacketData<UnknownEncrypted>>,
+    inbound_intro_packet: Option<PacketData<AssymetricEncrypted>>,
```

Comment:
I don't think when we compare the intro_packet we care about what type is it so don't decrypt it before, so this one probably needs to be UnknownEncrypted too